ql/src/java/org/apache/hadoop/hive/ql/exec/DataSketchesFunctions.java

```diff
@@ -128,14 +141,26 @@ private void buildCalciteFns() {
           OperandTypes.family(),
           unionFn);
 
+
       unionSFD.setCalciteFunction(unionFn);
       sketchSFD.setCalciteFunction(sketchFn);
+      if (estimateSFD != null) {
+        SqlFunction estimateFn = new HiveSqlFunction(estimateSFD.name,
+            SqlKind.OTHER_FUNCTION,
+            ReturnTypes.explicit(SqlTypeName.DOUBLE),
```

Comment:
If this is a UDF, we should probably dynamically generate the return type from it as we do for other UDFs?

Reply:
it's a little bit more complicated than what we have at other places:

* this should be run without anything being initialized - which means to get a "RexBuilder" I will need to construct that from the ground up....
* after that I could start registering; but to do that I would need a FuinctionInfo which is created only after the function is "registered" 
* and lastly: I have no info about what argument types the function is expecting (and how many) - when we construct these kind of things for other function we do have that...

I somehow feel like I'm locked into a specific thinking right now

Reply:
I've approached this right now by identifying the return type from the UDF class; we might need to replace that later - but it will work for now...

Reply:
Do we have a follow-up? Would you mind to create it so we can keep track?